flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/scala/expressionDsl.scala

```diff
@@ -1029,6 +1029,29 @@ object temporalOverlaps {
     TemporalOverlaps(leftTimePoint, leftTemporal, rightTimePoint, rightTemporal)
   }
 }
+/**
+  * Adds a (signed) integer interval to a timestamp. The unit for the interval is given
+  * by the unit argument, which should be one of the following values: "SECOND", "MINUTE",
+  * "HOUR", "DAY", "WEEK", "MONTH", "QUARTER" or "YEAR".
+  *
+  * e.g. timestampAdd("WEEK", 1, '2003-01-02'.toDate) leads to "2003-01-09".
+  */
+object timestampAdd {
+
+  /**
+    * Adds a (signed) integer interval to a timestamp. The unit for the interval is given
+    * by the unit argument, which should be one of the following values: "SECOND", "MINUTE",
+    * "HOUR", "DAY", "WEEK", "MONTH", "QUARTER" or "YEAR".
+    *
+    * e.g. timestampAdd("WEEK", 1, '2003-01-02'.toDate) leads to "2003-01-09".
+  */
+  def apply(
+      unit: Expression,
```

Comment:
Sorry for not having a look at this PR earlier.
The Table API expresses time intervals a bit differently than in this function. For example the interval of a tumbling window is defined as `window(Tumble over 5.milli on 'rowtime as 'w)`.
I think we should follow that style for consistency, i.e., `timestampAdd(1.week, 2003-01-02'.toDate)`.

Reply:
This will also ease the validation tests, because `1.week` is already a time interval type.

Reply:
thanks for your review and advice, @fhueske , yes, At the beginning I had thought these two kinds of function signature, and chose the current one following the sql signature. I think your suggestions is reasonable, it follows the table api style and  makes the code concise. 
There are two problems that I found are: one is that `1.quarter` has been realized in other meaning (Quarter table api) and another is `1.week` seems to be not existed right now (maybe we can add).Thank you

Reply:
I think we can and should add `week`. 

`quarter` is a bit more tricky... IMO, the current method is not well aligned with the other methods (`hour`, `day`, etc.) and it would make sense to change it for better consistency. However, such as change would break the API. What do you think @twalthr?

Reply:
+1 for better consistency. It is good to follow the Table-api style.

Reply:
+1 for this approach that directly specifies the interval literals. 

Regarding Quarter. It seems like a very old implementation and we should probably use `"1970-01-01".toDate.extract(TimeIntervalUnit.QUARTER)` to make it consistent with all other time unit extractions. What do you guys think?

I just tried it out by modifying the `Extract` method and it seems working perfectly.

Reply:
I agree with @walterddr but let's wait for @twalthr who added the `quarter` function

Reply:
I agree, we should use `quarter` for expressing a time and not for a special extraction function that can also be achieved with `.extract(...)`.